common/events.hpp

```diff
@@ -109,12 +125,13 @@ class Timer : public EventLoopObject {
 private:
 #ifdef MENDER_USE_BOOST_ASIO
 	asio::steady_timer timer_;
+	shared_ptr<bool> destroying_;
```

Comment:
Q: Why is a `shared_ptr` needed? :thinking:

Reply:
Because the handler in `AsyncWait` needs to keep its own copy, and this needs to survive the destruction of the timer, since Boost may call it late.